### PR TITLE
#151980425 chat

### PR DESCRIPTION
**Pull request by williamolojede**
#### What does this PR do?
add chat to game screen
#### Description of Task to be completed?
N/A

#### How should this be manually tested?
start a game and click on chat icon

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
#151980425

#### Screenshots (if appropriate)
N/A

#### Questions:
N/A

### CODE REVIEW

File: public/js/controllers/chat.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  }
+}
+
+angular.module('mean.system')
+  .controller('ChatController', ['$scope', '$firebaseArray', 'game', ($scope, $firebaseArray, game) => {
+    $scope.content = '';
+    $scope.isChatOpen = false;
+    $scope.hasUnreadMessages = false;
+
+    $scope.$watch('game.gameID', (newValue, oldValue) => {
```
[the remark this answers is not in the document]
Comment (andela-moe) [reply public/js/controllers/chat.js]: Can you please respond to this. `oldValue`

File: public/js/controllers/chat.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    $scope.addChat = () => {
+      const { playerIndex, players } = game;
+      const { avatar, username } = players[playerIndex];
+
+      if ($scope.content) {
+        const chat = new Chat({ avatar, username }, $scope.content);
+        $scope.chats.$add(chat);
+
+        const objDiv = document.querySelector('.chats');
```
Comment (andela-moe) [inline public/js/controllers/chat.js]: `objDiv` Is not a descriptive name for a variable. Please address this

File: public/js/controllers/chat.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    $scope.toggleChatOpen = () => {
+      // if chat box is open set has unread to false and clear text input
+      if ($scope.isChatOpen) {
+        $scope.hasUnreadMessages = false;
+        $scope.content = '';
+      }
+      $scope.isChatOpen = !$scope.isChatOpen;
+      const objDiv = document.querySelector('.chats');
+      // 50 is the bottom margin on the last li in chat list
+      objDiv.scrollTop = objDiv.scrollHeight + 50;
```
Comment (andela-moe) [inline public/js/controllers/chat.js]: `objDiv` Is not a descriptive name for a variable. Please address this

File: public/views/app.html [hunk 4]
```diff
@@ -72,7 +72,36 @@
...
+                      <span class="chat__user-name">{{chat.user.username}}</span>
+                      <span class="chat__created-at">{{chat.createdAt}}</span>
+                    </div>
+                    <p class="chat__message">{{chat.content}}</p>
+                  </div>
+                </li>
+              </ul>
+              <form ng-submit="addChat()" class="chat-box__form">
+                <img src="https://image.flaticon.com/icons/png/512/42/42877.png" alt="">
+                <input type="text" placeholder="donated? tell people about it" ng-model="content" id="chatInput"/>
```
Comment (andela-moe) [inline public/views/app.html]: Why do we have this "**donated? tell people about it**" text here

Comment (williamolojede) [reply public/views/app.html]: during our last demo, the product owner mentioned something about players talking about how much they've just donated to motivate others to donate also.

**Pull request closed without merging**